modules/volunteers/admin/sitesupport/editHours.js

```diff
@@ -0,0 +1,116 @@
+/* jshint esversion: 6 */
+/* globals apiRequest, confirmbox, showSidebar, simpleObjectToRequest */
+
+export default {
+  data() {
+    return {
+      record: {
+        hours: 0,
+        member: {
+          first_name: '',
+        },
+        enterer: {
+          id: '',
+        },
+        authorizer: {
+          id: '',
+        },
+        department: {
+          name: '',
+        },
+        end: null,
+      },
+    }
+  },
+  methods: {
+    commitHours() {
+      confirmbox(
+        'Please! double check entries',
+        'Proceed with Volunteer Hour Update?').then(() => {
+        var item = {};
+
+        item['hours'] = parseFloat(this.record.hours);
+        item['end'] = this.record.end;
+        var e = document.getElementById('edit_mod');
+        item['modifier'] = e.options[e.selectedIndex].value;
+        item['department'] = this.record.department.id;
+        item['authorizer'] = this.record.authorizer.id;
+
+        apiRequest('PUT', '/volunteer/hours/' + this.record.id + '?force=1', simpleObjectToRequest(item))
+          .then(() =>  {
+            location.reload();
+          });
+      });
+    },
+    deleteHours() {
+      confirmbox('DELETE Volunteer Entry?').then(() => {
+        apiRequest('DELETE', '/volunteer/hours/' + this.record.id)
+          .then(() => {
+            location.reload();
+          });
+      });
+    },
+    show(record) {
+      /* deep copy */
+      this.record = JSON.parse(JSON.stringify(record));
```

Comment:
Can you help me understand why we need to deep copy `record` here?

Reply:
The deep copy is required so that we do not edit the record passed to us from the table itself. Otherwise as we change values the change live on that record and we would require robust undo logic to resort the state if the edit sidebar is closed or canceled.

Reply:
When you say "from the table itself", do you mean the table component defined in `hourTable` or do you mean like a database table record?

Reply:
Yes, from the hourTable.

Reply:
And to make sure I understand - we _don't_ want to track those changes if the sidebar has closed? Is that because this component itself has been setup to post its own data to the server?

Reply:
Correct